lib/dma.c

```diff
@@ -536,16 +536,17 @@ dma_controller_dirty_page_logging_stop(dma_controller_t *dma)
 #ifdef DEBUG
 static void
 log_dirty_bitmap(vfu_ctx_t *vfu_ctx, dma_memory_region_t *region,
-                 char *bitmap, size_t size)
+                 char *bitmap, size_t size, size_t pgsize)
 {
     size_t i;
     size_t count;
     for (i = 0, count = 0; i < size; i++) {
-        count += __builtin_popcount(bitmap[i]);
+        count += __builtin_popcount((uint8_t)bitmap[i]);
```

Comment:
why this change?

Reply:
I think this was actually wrong to begin with (maybe this change should be a separate PR then?).

Without the cast, when you run
```bash
LIBVFIO_SO_DIR='/path/to/libvfio-user/build/lib' pytest-3 test/py/test_dirty_pages.py --capture=no
```
the third call to get the dirty bitmap logs this line:
```
DEBUG: dirty pages: get [0x10000, 0x30000), 27 dirty pages of size 4096
```
which is wrong, as the test correctly asserts that the bitmap equals `0b0000001111000001` (5 dirty pages).

When you add the cast, it correctly reports that there were 5 dirty pages.

I imagine that, because `__builtin_popcount` takes an `unsigned int`, there is no implicit conversion going on (or if there is, it's doing something wrong), so some of the bits in the bitmap are being counted multiple times.

What do you think? @jlevon

Reply:
Separate PR please. The bug is that the (signed) char is getting sign-extended on conversion to the larger type.

Reply:
https://github.com/nutanix/libvfio-user/pull/766